src/main/java/Dog.java

```diff
@@ -0,0 +1,27 @@
+public class Dog extends Animal{
+    int barrierRun = 500;
+    int barrierSwim  = 10;
+    @Override
+    void Run(int path) {
+        super.Run(path);
+        if (path >= barrierRun){
```

Comment:
Those `ifs` may go to the parent class. Logic is fully the same

Reply:
Would be nice to discuss with you to understand what you mean